infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/RoutingQueuesSnapshotReader.java

```diff
@@ -26,4 +66,167 @@ public void reset(long ts) {
         // In addition to setting the snapshot timestamp to ts, write to the table subscribed to by the client
         // requesting for a snapshot sync
     }
+
+    @Override
+    public SnapshotReadMessage read(UUID syncRequestId) {
+        List<LogReplication.LogReplicationEntryMsg> messages = new ArrayList<>();
+
+        LogReplication.LogReplicationEntryMsg msg;
+
+        log.info("Start Snapshot Sync replication for stream name={}, id={}", streamToReplicate,
+            CorfuRuntime.getStreamID(streamToReplicate));
+
+        // If the current opaque stream has data, read and send it
+        if (currentStreamHasNext()) {
```

Comment:
`currentStreamInfo` can be null when we come here...we need a null check and assignment of `currentStreamInfo` similar to the baseSnapshotReader.read()